src/postgres/src/interfaces/libpq/fe-connect.c

```diff
@@ -748,6 +810,611 @@ PQconnectStartParams(const char *const *keywords,
+			(conn->pghostaddr && strcmp(conn->pghostaddr,PQgetvalue(res, itr , 0))==0))
+			{
+				YBclientNetworkStatus=1;
+				break;
+			}
+		}
+	}
+
+	/*
+	 * If the Network status is not yet decided.
+	 * Identify if the Client is outside the Private network.
+	 */
+	if (YBclientNetworkStatus==0)
+	{
+		int itr;
+	 	for(itr=0; itr<nServers; itr++)
+		{
+			if((conn->pghost && strcmp(conn->pghost,PQgetvalue(res, itr , 7))==0) ||
+			(conn->pghostaddr && strcmp(conn->pghostaddr,PQgetvalue(res, itr , 7))==0))
+			{
+				YBclientNetworkStatus=-1;
+				break;
+			}
+		}
+	}
+
+	/*
+	 * If the Network status is not yet decided.
+	 * Try connecting with all the Private IP address.
+	 */
+	if (YBclientNetworkStatus==0)
+	{
+		int itr;
+		for(itr=0; itr<nServers; itr++)
+		{
+			if(YBtestNetwork(PQgetvalue(res, itr , 0)))
+			{
+				YBclientNetworkStatus=1;
+				break;
+			}
+		}
+	}
+
+	/*
+	 * Since the Client was unable to connect with the Private ips.
+	 * It must not have access to it. Thus we need to make the connection
+	 * to with the public ip.
+	 */
+	if (YBclientNetworkStatus==0)
+		YBclientNetworkStatus=-1;
+
+	/*
+	 * Locking the map_ready so that no other thread can start iterating it.
+	 */
+	pthread_mutex_lock(&(map_ready));
+
+	/*
+	 * Update the yb_last_update_time
+	 */
+	yb_last_update_time = time(NULL);
+
+	/*
+	 * 1. Assign the value of is_running as false for all the servers.
+	 * 2. Iterate the result and update the is_running as true.
+	 * 3. Maintain the count of the servers which were not found in the server_details.
+	 * 4. If this count is not zero reallocate the map memory and add the values to it.  
+	 */
+
+	/*
+	 * Assign the value of is_running as false for all the servers.
+	 */
+	for(int i =0; i< total_servers; i++)
+	{
+		server_details[i].is_running = false;
+	}
+
+	/*
+	 * Iterate the result and update the is_running as true.
+	 */
+  	for (int i = 0; i < nServers; i++)
+  	{
+	  
+		char  *server;
+		if(YBclientNetworkStatus==1) 
+			server= PQgetvalue(res, i , 0);	
+		else 
+			server= PQgetvalue(res, i , 7);
+		
+		/*
+		 * Since no other thread is iterating the server_details (map_ready is locked),
+		 * YBserverStatusChange can be called without considering the 
+		 * thread_safe (keeping the value of check_thread_safe as false).
+		 * If kept true will endup in a deadlock.
+		 */
+		if (!YBserverStatusChange(server,true,false))
+		{
+			/*
+		 	* Maintain the count of the servers which were not found in the map.
+		 	*/
+			increase_map_size++;
+			server_to_add[i] = true;
+		}else
+			server_to_add[i] = false;
+
+	}
+
+	/*
+	 * 4. If count is not zero reallocate the map memory and add the values to it.  
+	 */
+	if(increase_map_size != 0)
+	{
+		/*
+		 * Allocate the space
+		 */
+		struct node_details *temp_server_details = (struct node_details *) malloc ( (increase_map_size + total_servers) * sizeof (struct node_details));
+		for(int i =0; i < total_servers; i++)
```

Comment:
Where do you handle the servers which are no longer available?

Reply:
For the servers that are not running, the "is_running" flag will be set to "0".
Once the server starts again, the "is_running" flag will be set to "1" again.